transmart/api/v2/query.py

```diff
@@ -42,12 +45,24 @@ def headers(self):
         return {'Accept': 'application/{};charset=UTF-8'.format('hal+json' if self.hal else 'json')}
 
     def get_constraints(self):
-        constraints = ''.join([str(c) for c in (self.in_study, self.in_patientset, self.in_concept) if c.value])
-
-        if constraints:
-            return {'constraint': constraints}
+        constraints = []
+        for c in (self.in_study, self.in_patientset, self.in_concept):
+            if isinstance(c, list):
+                for v in c:
+                    constraints.append(v.get_constraint())
+            elif c.value:
+                constraints.append(c.get_constraint())
+
+        if len(constraints) > 1:
+            constraints = {'constraint': json.dumps({"type" : self.operator, "args": constraints})}
+        elif len(constraints) == 1:
+             if self.handle == "/v2/patient_sets":
```

Comment:
why is this special case needed?

Reply:
For some reason the way a constraint is handled by v2/patient_sets is different than the other API calls. See https://github.com/thehyve/transmart-core/blob/b4f5fd7975b03546631f33da609ccad471be7cbb/transmart-rest-api/grails-app/controllers/org/transmartproject/rest/PatientQueryController.groovy#L168